test/benchmark/api_v1_1_tests.py

```diff
@@ -0,0 +1,137 @@
+from tests.api.v1_1.discovery_tests import DiscoveryTests
+from tests.api.v1_1.poller_tests import PollerTests
+
+from benchmark import ansible_ctl
+from benchmark.utils import parser
+from benchmark.utils.case_recorder import caseRecorder
+
+LOG = Log(__name__)
+
+class BenchmarkTests(object):
+    def __init__(self, name):
+        ansible_ctl.render_case_name(name)
+        self.__data_path = ansible_ctl.get_data_path_per_case()
+        self.case_recorder = caseRecorder(self.__data_path)
+        self.client = config.api_client
+        self.node_count = 0
+
+    def prepare_case_env(self):
+        self.node_count = self.check_compute_count()
+        self.case_recorder.write_interval(ansible_ctl.get_data_interval())
+        self.case_recorder.write_start()
+        self.case_recorder.write_node_number(self.node_count)
+
+        assert_equal(True, ansible_ctl.start_daemon(), \
+                    message='Failed to start data collection daemon!')
+
+    def collect_case_data(self):
+        assert_equal(True, ansible_ctl.collect_data(), message='Failed to collect footprint data!')
+        self.case_recorder.write_end()
+
+        # parser.parse(self.__data_path)
+
+    def check_compute_count(self):
+        Nodes().nodes_get()
+        nodes = loads(self.client.last_response.data)
+        count = 0
+        for n in nodes:
+            type = n.get('type')
+            if type == 'compute':
+                count += 1
+        return count
+
+
+@test(groups=["benchmark.poller"])
+class BenchmarkPollerTests(BenchmarkTests):
+    def __init__(self):
+        BenchmarkTests.__init__(self,'poller')
+
+    @test(groups=["test-bm-poller"], depends_on_groups=["test-node-poller"])
+    def test_poller(self):
+        """ Wait for 15 mins to let RackHD run pollers """
+        self.prepare_case_env()
+        time.sleep(900)
+        self.collect_case_data()
+        LOG.info('Fetch poller log finished')
+
+@test(groups=["benchmark.discovery"])
+class BenchmarkDiscoveryTests(BenchmarkTests):
+    def __init__(self):
+        BenchmarkTests.__init__(self,'discovery')
+        self.__task = None
+        self.__discovered = 0
+
+    @test(groups=["test-bm-discovery-prepare"], depends_on_groups=["test-node-poller"])
+    def test_prepare_discovery(self):
+        """ Prepare discovery """
+        self.prepare_case_env()
+
+    @test(groups=["test-bm-discovery"],
+            depends_on_groups=["test-bm-discovery-prepare", "test_discovery_delete_node"])
+    def test_discovery(self):
+        """ Wait for discovery finished """
+        self.case_recorder.write_event('start all discovery')
+        self.__task = WorkerThread(AMQPWorker(queue=QUEUE_GRAPH_FINISH, \
+                                              callbacks=[self.handle_discovery_finish]), \
+                                   'benchmark discovery')
+        def start(worker, id):
+            worker.start()
+        tasks = WorkerTasks(tasks=[self.__task], func=start)
```

Comment:
Does it possible to replace `func=start` with `func = lambda w: w.start()` , so that we could omit the definition `def start(worker, id):`